### DISCUSSION TITLE
operators import error

### DISCUSSION THREAD

**Discussion by praveen1219** [apache/airflow#24917] [General] [answered]
### Apache Airflow Provider(s)

mongo, mysql

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==3.3.0
apache-airflow-providers-cncf-kubernetes==4.0.0
apache-airflow-providers-ftp==2.1.2
apache-airflow-providers-http==2.1.2
apache-airflow-providers-imap==2.2.3
apache-airflow-providers-microsoft-azure==3.8.0
apache-airflow-providers-mongo==3.0.0
apache-airflow-providers-mysql==2.2.3
apache-airflow-providers-postgres==4.1.0
apache-airflow-providers-sqlite==2.1.3

### Apache Airflow version

2.3.2 (latest released)

### Operating System

ubuntu v 20.04

### Deployment

Virtualenv installation

### Deployment details

installed in virtual environament

### What happened

Recently dummy operator under airflow.providers.dummy is depricated and changed to empty operator.
But while importing empty operator it says (from airflow.operators.empty import EmptyOperator
ModuleNotFoundError: No module named 'airflow.operators.empty') 

### What you think should happen instead

It should be imported and used without error

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment, marked as answer]
You probably have your environment broken. But how. I have no idea. You (very likely) wrongly specified the set of packages you have installed and probably forced installation wiht ignoring dependencies. You shuold follow the `only supported` way of installing airlfow (with constraints https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html) and reinstall your airflow to correct broken installation you have.

Also from your description It looks like you actually have two versions of airflow installed in your system (an old one and new one) and that sometimes the old version is used and sometimes new one which indicates that your deployment is completely broken and you need to redo it from scratch. Which I would heartily recommend, because otherwise you will have to spend a lot of time on tracing what's wrong with it and where you made mistake (for example it can happen if you have multiple virtualenvs, or when you installed airflow globally AND with --user flag etc,. etc. This is not an airflow problem - this problem with your environment, that only you can diagnose and fix and no-one else can help you with it other than pointing out to potential problems (like I did now).

If it does not help, please document exactly which version of airflow you have installed and how you installed them and show the exact logs of your import errors and we might be able to help to point what mistake you've done when installing Airlfow. But you will need ot spend quite some time on gathering all details - what are your installation directories of your python packages (there are many), where airflow packages are installed and so on.
